src/array.jl

```diff
@@ -464,11 +471,10 @@ end
 copyto!(dest::CatArrOrSub, src::CatArrOrSub) =
     copyto!(dest, 1, src, 1, length(src))
 
-copyto!(dest::CatArrOrSub, dstart::Integer, src::CatArrOrSub) =
-    copyto!(dest, dstart, src, 1, length(src))
-
 @static if VERSION >= v"0.7.0-DEV.3208"
     using Future
+    Future.copy!(dest::CatArrOrSub, src::AbstractArray) =
+        copyto!(dest, 1, src, 1, length(src))
     Future.copy!(dest::CatArrOrSub, src::CatArrOrSub) =
```

Comment:
This method isn't needed now AFAICT.

Reply:
What do you want to happen if we want to `copy!` a normal array to a categorical array? Throw an error?